### DISCUSSION TITLE
Giving up on Celery BUGS collection

### DISCUSSION THREAD

**Discussion by Ixiodor** [celery/celery#7017] [Q&A]
I know it'll be closed but, I can't handle 1 BUG patched and 10 new BUGS introduced, so I'm going to give up Celery probably. Those are new BUGS I found that want report for your future progress(I'm on 5.0.2 from source actually cause it has my PR that i needed):

1. Stopping workers will mark all prefetched task as FAILED, losing them.(#6465 )
2. Workers process pool will decrase without reasons while working, used to start with -c 4 and noticed that fall to 1/2 till restart.(#6467)
3. Celery control.active collects only workers with empty tasks. As a worker gets tasks, you'll lose it.(#6466 )

**Comment by thedrow** [comment]
I'm sorry that you feel that way.
We're fully aware of the status of this project and we're working to make it better.

Do you have the issue numbers we need to prioritize?

**Comment by Ixiodor** [comment]
I did some PR when I found BUGS but seems out of control. Actually from my list, point 1 and 2 are the killing one. Compromises the base usage of Celery.

**Comment by thedrow** [comment]
If you'd like us to fix the problems we need to know which are the most urgent.
Are these issues which were not reported to us as of yet?
If not, please provide issue numbers.

**Comment by Ixiodor** [comment]
Made a BUG report for most important one

**Comment by thedrow** [comment]
And the rest?

**Comment by Ixiodor** [comment]
Updated.

**Comment by matusvalo** [comment]
I suppose that we need to improve tests of Celery because otherwise new bugs will pop out again and again... Mainly for such difficult code-base as Celery has.

**Comment by auvipy** [comment]
> I know it'll be closed but, I can't handle 1 BUG patched and 10 new BUGS introduced, so I'm going to give up Celery probably. Those are new BUGS I found that want report for your future progress(I'm on 5.0.2 from source actually cause it has my PR that i needed):
> 
>     1. Stopping workers will mark all prefetched task as FAILED, losing them.(#6465 )
> 
>     2. Workers process pool will decrase without reasons while working, used to start with -c 4 and noticed that fall to 1/2 till restart.(#6467)
> 
>     3. Celery control.active collects only workers with empty tasks. As a worker gets tasks, you'll lose it.(#6466 )

we are sorry you felt that way. distributed programming, in general, is a tough domain. as celery provides many things and the code base is big it's natural to have some edge cases which need a lot of effort & time to track & fix.
